### ISSUE TITLE
Improve Usability of the `OSCALEditableTextField` 

### ISSUE THREAD

**Issue by kylelaker** [EasyDynamics/oscal-react-library#270]
#As a user of the OSCAL Editor, I want to be able to save or cancel changes via the keyboard so that I can edit the field without a mouse for speed/convenience and potentially accessibility reasons.

When operating in "Rest mode", pressing the `Enter` key in an `OSCALEditableTextField` does not "submit" the change. A user is required to click the save or cancel button.

Pressing "Enter" should properly save that particular field and end editing, as if the "save" button had been pressed. Additionally, pressing the "Esc" should exit editing _without_ saving any changes as if the "Cancel" button had been pressed. Both the buttons should have a Tooltip on hover to clarify their purpose and which field they apply to (simiar to the existing `aria-label` logic).

### Acceptance Criteria
 - Within an `OSCALEditableTextField` pressing "Enter" saves the field, as if the Save button had been clicked
 - Within an `OSCALEditableTextField` pressing "Esc" exits editing of the field, as if the Cancel button had been clicked
 - All iconography in an `OSCALEditableTextField` has a tooltip or similar element (such as perhaps `title` or `<label>`) to clarify the purpose of that button
 - `aria-label` attributes contain human-readable text within `OSCALEditableTextField` and its children
 - End-to-End (Cypress) tests have been created

### Implementation Notes
 - The text field itself is a `@material-ui/core/TextField` which may require using the `onKeyPress` event since the elements are not an `Input` within a `<form>`.
 - We want to avoid copying/pasting code, ideally we'll find a way to implement this functionally by more directly invoking the underlying implementation rather than copying `onClick()` code to an `onKeyPress()` function

### Subtasks
- [x] #424
- [x] #431
- [x] #441
- [x] #442
- [x] #443
- [x] (https://github.com/EasyDynamics/oscal-editor-deployment/issues/97)
